Libraries/Animated/src/AnimatedImplementation.js

```diff
@@ -1508,21 +1510,58 @@ function createAnimatedComponent(Component: any): any {
 
     componentWillUnmount() {
       this._propsAnimated && this._propsAnimated.__detach();
+      this._detachNativeEvents(this.props);
     }
 
     setNativeProps(props) {
       this._component.setNativeProps(props);
     }
 
     componentWillMount() {
-      this.attachProps(this.props);
+      this._attachProps(this.props);
     }
 
     componentDidMount() {
       this._propsAnimated.setNativeView(this._component);
+
+      this._attachNativeEvents(this.props);
     }
 
-    attachProps(nextProps) {
+    _attachNativeEvents(newProps) {
+      if (newProps !== this.props) {
+        this._detachNativeEvents(this.props);
+      }
+
+      // Make sure to get the scrollable node for components that implement
+      // `ScrollResponder.Mixin`.
+      const ref = this._component.getScrollableNode ?
+        this._component.getScrollableNode() :
+        this._component;
+
+      for (const key in newProps) {
+        const prop = newProps[key];
+        if (prop instanceof AnimatedEvent && prop.__isNative) {
+          prop.__attach(ref, key);
+        }
+      }
+    }
+
+    _detachNativeEvents(props) {
+      // Make sure to get the scrollable node for components that implement
+      // `ScrollResponder.Mixin`.
+      const ref = this._component.getScrollableNode ?
```

Comment:
This only works for direct events, pan responder does add some JS code so we'd have to create some sort of native pan responder if we want to support this. It should be possible to use Animated.event on the native touch move event directly though.

This is just so it works properly with ScrollView which sometimes is wrapped with RefreshControl. We have to use this method to make sure to always get a ref to the native ScrollView.

I havent tested with other events than scrollview onScroll but it is generic enough that it should work, I'll test with touch move on a view today to make sure it works.